mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulEditProfileDialog.kt

```diff
@@ -0,0 +1,62 @@
+package ch.epfl.sdp.mobile.state
+
+import androidx.compose.runtime.*
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
+import ch.epfl.sdp.mobile.ui.setting.EditProfileNameDialog
+import ch.epfl.sdp.mobile.ui.setting.EditProfileNameDialogState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+/**
+ * Implementation of the [EditProfileNameDialogState] interface
+ *
+ * @param user the current [AuthenticatedUser].
+ * @param scope the coroutine scope.
+ * @param onSave the callback called after we saved the changes.
+ * @param onCancle the callback called when we click the cancle button.
+ */
+class EditProfileNameDialogStateImpl(
+    user: AuthenticatedUser,
+    private val scope: CoroutineScope,
+    onSaveAction: State<() -> Unit>,
+    onCancleAction: State<() -> Unit>
+) : EditProfileNameDialogState {
+  var user by mutableStateOf(user)
+  val onSaveAction by onSaveAction
+  val onCancleAction by onCancleAction
+
+  override var name by mutableStateOf(user.name)
+
+  override fun onSaveClick() {
+    scope.launch {
+      user.update { name(name) }
+      onSaveAction()
+    }
+  }
+
+  override fun onCancleClick() {
+    onCancleAction()
+  }
+}
+
+/**
+ * A stateful implementation of [EditProfileNameDialog] which uses some composition-local values to
+ * retrieve the appropriate dependencies.
+ *
+ * @param user the current [AuthenticatedUser].
+ * @param onSave the callback called after we saved the changes.
+ * @param onCancle the callback called when we click the cancle button.
+ */
+@Composable
+fun StatefulEditProfileDialog(user: AuthenticatedUser, onSave: () -> Unit, onCancle: () -> Unit) {
```

Comment:
This is where you'll want to provide a single `onClose` action.

Reply:
As explained previously, I want to make the interface to the Dialog similar to the other dialog we have.